tests/integration/test_fim/test_files/conftest.py

```diff
@@ -4,38 +4,88 @@
 from wazuh_testing.tools.monitoring import FileMonitor
 from wazuh_testing.tools.services import control_service
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope="module")
 def restart_syscheckd(get_configuration, request):
     """
     Reset ossec.log and start a new monitor.
     """
-    control_service('stop', daemon='wazuh-syscheckd')
+    control_service("stop", daemon="wazuh-syscheckd")
     truncate_file(LOG_FILE_PATH)
     file_monitor = FileMonitor(LOG_FILE_PATH)
-    setattr(request.module, 'wazuh_log_monitor', file_monitor)
-    control_service('start', daemon='wazuh-syscheckd')
+    setattr(request.module, "wazuh_log_monitor", file_monitor)
+    control_service("start", daemon="wazuh-syscheckd")
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope="function")
+def restart_syscheckd_function(get_configuration, request):
+    """
+    Restart syscheckd daemon.
+    """
+    control_service("stop", daemon="wazuh-syscheckd")
+    truncate_file(LOG_FILE_PATH)
+    file_monitor = FileMonitor(LOG_FILE_PATH)
+    setattr(request.module, "wazuh_log_monitor", file_monitor)
+    control_service("start", daemon="wazuh-syscheckd")
+
+
+@pytest.fixture(scope="module")
 def wait_for_fim_start(get_configuration, request):
     """
     Wait for realtime start, whodata start or end of initial FIM scan.
     """
-    file_monitor = getattr(request.module, 'wazuh_log_monitor')
-    mode_key = 'fim_mode' if 'fim_mode2' not in get_configuration['metadata'] else 'fim_mode2'
+    file_monitor = getattr(request.module, "wazuh_log_monitor")
+    mode_key = (
+        "fim_mode" if "fim_mode2" not in get_configuration["metadata"] else "fim_mode2"
+    )
 
     try:
-        if get_configuration['metadata'][mode_key] == 'realtime':
+        if get_configuration["metadata"][mode_key] == "realtime":
             detect_realtime_start(file_monitor)
-        elif get_configuration['metadata'][mode_key] == 'whodata':
+        elif get_configuration["metadata"][mode_key] == "whodata":
             detect_whodata_start(file_monitor)
         else:  # scheduled
             detect_initial_scan(file_monitor)
     except KeyError:
         detect_initial_scan(file_monitor)
+
+
+@pytest.fixture(scope="function")
+def wait_for_fim_start_function(get_configuration, request):
+    """
+    Wait for realtime start, whodata start or end of initial FIM scan.
+    """
+    file_monitor = getattr(request.module, "wazuh_log_monitor")
+    mode_key = (
+        "fim_mode" if "fim_mode2" not in get_configuration["metadata"] else "fim_mode2"
+    )
+
+    try:
+        if get_configuration["metadata"][mode_key] == "realtime":
+            detect_realtime_start(file_monitor)
+        elif get_configuration["metadata"][mode_key] == "whodata":
+            detect_whodata_start(file_monitor)
+        else:  # scheduled
+            detect_initial_scan(file_monitor)
+    except KeyError:
+        detect_initial_scan(file_monitor)
+
```

Comment:
Talking with Seyla we decide to maintenance two fixtures but maybe you can defined a new function with the code inside the fixture in order to not duplicated the code.

For example:
```suggestion
@pytest.fixture(scope="function")
def wait_for_fim_start_function(get_configuration, request):
   wait_for_fim_start(get_configuration, request)


def wait_for_fim_start(get_configuration, request):
    """
    Wait for realtime start, whodata start or end of initial FIM scan.
    """
    file_monitor = getattr(request.module, "wazuh_log_monitor")
    mode_key = (
        "fim_mode" if "fim_mode2" not in get_configuration["metadata"] else "fim_mode2"
    )

    try:
        if get_configuration["metadata"][mode_key] == "realtime":
            detect_realtime_start(file_monitor)
        elif get_configuration["metadata"][mode_key] == "whodata":
            detect_whodata_start(file_monitor)
        else:  # scheduled
            detect_initial_scan(file_monitor)
    except KeyError:
        detect_initial_scan(file_monitor)
```

Reply:
Created the `wait_for_fim_active` function, that will be called by both wait_for_fim_scan fixtures.
Take note that the` wait_for_fim_scan` fixture with `module scope` should be renamed, but it appears in 263 places in the repo, so that refactor must be taken with care and is beyond the scope of this PR.